pkg/cmd/roachtest/cluster.go

```diff
@@ -1393,6 +1394,61 @@ func (c *clusterImpl) FetchDebugZip(
 	})
 }
 
+// FetchVMSpecs downloads the VM specs from the cluster using `roachprod get`.
+// The logs will be placed in the test's artifacts dir.
+func (c *clusterImpl) FetchVMSpecs(ctx context.Context, l *logger.Logger) error {
+	if c.IsLocal() {
+		return nil
+	}
+
+	l.Printf("fetching VM specs\n")
+	c.status("fetching VM specs")
+
+	vmSpecsFolder := filepath.Join(c.t.ArtifactsDir(), "vm_specs")
+	if err := os.MkdirAll(vmSpecsFolder, 0755); err != nil {
+		return err
+	}
+
+	// Don't hang forever if we can't fetch the specs.
+	return timeutil.RunWithTimeout(ctx, "fetch logs", 5*time.Minute, func(ctx context.Context) error {
+		cachedCluster, err := getCachedCluster(c.name)
+		if err != nil {
+			return err
+		}
+		providerToVMs := bucketVMsByProvider(cachedCluster)
+
+		var vmSpecs []map[string]interface{}
+		for provider, vms := range providerToVMs {
+			p := vm.Providers[provider]
+			vmSpecs, err = p.GetVMSpecs(l, vms)
```

Comment:
This would overwrite `vmSpecs` on each iteration, you should combine the maps instead.

Reply:
Updated the logic

Reply:
Apologies if you're still working on it, but I'm still getting only one file per run 🤔

Reply:
Yes, still working on it. Not sure what's going wrong in multinode tests.

Reply:
I think it's because you're returning here:

https://github.com/cockroachdb/cockroach/blob/33a15640be94b0167031d979e1c4f4b708a4acbb/pkg/cmd/roachtest/cluster.go#L1440

Reply:
Ah I just had to something like 🤦🏻‍♂️
```
err = os.WriteFile(dest, specJSON, 0644)
if err != nil {
         l.Printf("Failed to write spec to file for %s\n", name)
	continue
}
```

Reply:
Worked 🥳